controllers/usersignup/usersignup_controller.go

```diff
@@ -655,7 +657,14 @@ func (r *Reconciler) ensureNewMurIfApproved(reqLogger logr.Logger, config toolch
 	}
 
 	// Provision the MasterUserRecord
-	return r.provisionMasterUserRecord(reqLogger, config, userSignup, targetCluster, userTier)
+	if err := r.provisionMasterUserRecord(reqLogger, config, userSignup, targetCluster, userTier); err != nil {
+		return err
+	}
+	// track activation in Segment
+	if r.SegmentClient != nil {
```

Comment:
Instead of reporting it here I think we should report it in provisionMasterUserRecord(). Right after or just before incrementing our user count metrics. Otherwise if there is a problem with provisioning a user you will report it multiple times.

Reply:
good point, let me refactor that!

Reply:
addressed in e701dec: done in the `updateUserSignupMetricsByState` func when metrics are updated